ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java

```diff
@@ -15,13 +15,64 @@
 import org.opensearch.ml.common.output.MLOutput;
 import org.opensearch.ml.common.output.Output;
 
+import java.nio.file.Path;
 import java.util.Map;
 
 /**
  * This is the interface to all ml algorithms.
  */
 public class MLEngine {
 
+    public static Path DJL_CACHE_PATH;
+    public static Path DJL_MODELS_CACHE_PATH;
+    public static void setDjlCachePath(Path opensearchDataFolder) {
+        DJL_CACHE_PATH = opensearchDataFolder.resolve("djl");
+        DJL_MODELS_CACHE_PATH = DJL_CACHE_PATH.resolve("models_cache");
+    }
+
+    public static Path getUploadModelPath(String modelId, String modelName, Integer version) {
+        return getUploadModelPath(modelId).resolve(version + "").resolve(modelName);
```

Comment:
Can version be null? if not it's better use int

Avoid using `(version + "")`, use version.toString() or `String.valueOf(version)` if version is primitive type.